tests/conftest.py

```diff
@@ -52,6 +54,47 @@ def loader(relpath, **kwargs):
     return loader
 
 
+def checkCarlaPath():
+    CARLA_ROOT = os.environ.get("CARLA_ROOT")
+    if not CARLA_ROOT:
+        pytest.skip("CARLA_ROOT env variable not set.")
+    return CARLA_ROOT
+
+
+@pytest.fixture
+def launchCarlaServer():
+    CARLA_ROOT = checkCarlaPath()
+    carla_process = subprocess.Popen(
+        f"bash {CARLA_ROOT}/CarlaUE4.sh -RenderOffScreen", shell=True
+    )
+    # NOTE: CARLA server takes time to start up
+    time.sleep(3)
+    yield
+    while carla_process.poll() is None:
+        carla_process.kill()
+        time.sleep(0.1)
+    # NOTE: CARLA server hangs. The first `kill` usually works,
+    # but for safe measure we needed to kill hanging zombie processes
```

Comment:
I don't understand this: `poll()` is not supposed to return a non-`None` value until the process has terminated, so how can the process still be running when we get here?

I'm also not sure if there's any value in repeatedly sending SIGKILL. We could just send SIGKILL, wait a few seconds, and then send SIGTERM (with `terminate()`) if the process is still open.